### PR TITLE
Theme: Try a `PatternGrid` component to display patterns

### PR DESCRIPTION
**Pull request by ryelle**
See #62 — This adds a few components to create a dynamic pattern grid. It uses the same structure and CSS introduced in #68, but also adds an iframe'd pattern preview.

This mainly gets the components in place, but doesn't touch anything data-y yet, not really. 

I'm using `select( coreStore ).getEntityRecords( 'postType', 'wporg-pattern' );` to get the list of patterns, but this generates an API request that only works for logged in editors. We'll want to create our own store to fetch just the plain `wporg-patterns` endpoint, not the edit context. See some discussion on https://github.com/WordPress/pattern-directory/issues/66#issuecomment-812450704.

❓ Would we consider this OK-enough to merge, or should we try to get the data side in first?

⚠️  I know the previews aren't great (design-wise), but I think I'd rather iterate on those in future PRs. Open to debate on that too 🙂 

### Screenshots

Screen Shot 2021-04-02 at 6 18 20 PM

### How to test the changes in this Pull Request:

0. Be logged in (see above re: API)
1. Load any archive page
2. It should render the list of patterns _(It will always show you the latest patterns, no filtering yet)_
3. Click on a pattern
4. It should take you to the pattern detail view, no change here

### CODE REVIEW

**Review by StevenDufresne — APPROVED** [review]
I'm fine with this being merged as-is. 

I'm a little skeptical of iframe usage at scale but we can adjust as the project matures.

File: public_html/wp-content/themes/pattern-directory/src/components/pattern-grid/index.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import PatternThumbnail from '../pattern-thumbnail';
+
+function PatternGrid() {
+	const posts = useSelect( ( select ) => {
+		// @todo This only works for logged in users, we'll need to create our own store for general use.
+		const { getEntityRecords } = select( coreStore );
+		return getEntityRecords( 'postType', 'wporg-pattern' );
+	} );
+	// `posts` will be null while the fetch happens.
+	const isLoading = ! Array.isArray( posts );
```
Comment (StevenDufresne) [inline public_html/wp-content/themes/pattern-directory/src/components/pattern-grid/index.js:20]: One thing to mention here is that `posts` will start as an empty array then switch to `null` when the query starts which makes sense for this component, but gives an intermediate state in the view which is more obvious in slower connections. It leads to some jank on load.

**Review by StevenDufresne — APPROVED** [review]

File: public_html/wp-content/themes/pattern-directory/src/components/pattern-grid/index.js (hunk 1 above)
Comment (ryelle) [reply public_html/wp-content/themes/pattern-directory/src/components/pattern-grid/index.js:20]: Yeah, this is mostly placeholder for a real data infrastructure, where we could have a different selector for the `isLoading` state.

**Comment by ryelle** [comment]
> I'm a little skeptical of iframe usage at scale but we can adjust as the project matures.

That's fair - we also have the option of images from mshots or e2e/puppeteer, or trying out a shadow DOM approach (I haven't used that before, but it sounds like it would have the same sandboxing effect as the iframe).

**Pull request merged**